### ISSUE TITLE
Overload check failed for property `__get__`

### ISSUE THREAD

**Issue by yanyongyu** [microsoft/pyright#6684]
**Describe the bug**

Pyright reports errors when calling property's `__get__` function.

**Code or Screenshots**

[playground](https://pyright-play.net/?code=MYGwhgzhAECCBcAoaLoAEAOAnA9hgplgC4CeyqAJvgGbRH4REAUE%2BI1AlNALQB80jLElQjoWfEQCuWAHbQARPUbzEiMNAC8cJh0SwAdEqL6A%2BiYDmEs0zC6gA)

```python
class A:
    @property
    def test(self) -> str:
        return "test"

a = A()
A.test.__get__(a)  # Error
```

Pright reports:

```
No overloads for "__get__" match the provided arguments
Argument of type "A" cannot be assigned to parameter "obj" of type "None" in function "__get__"
```

**VS Code extension or command-line**
Pyright 1.1.339, command-line

**Comment by erictraut** [comment]
Pyright has a bunch of special-case logic for `property`. Type checkers need to special-case it because the `property` class should have been defined as generic, but for historical reasons it was not. I will look at whether I can make the special-case logic work with your unusual use of the `__get__` method, but I may conclude that it cannot be fixed without breaking more typical use cases.

Out of curiosity, why are you using `A.test.__get__(a)` rather than `a.test`? The latter will work fine, and they should be identical in their runtime behavior.

**Comment by erictraut** [comment]
One option is to write your own descriptor class that is similar to `property` but is a proper generic class.
